apps/frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { type ReactNode, useCallback } from "react";
+
+import "./modal.css";
+
+interface ModalProperties {
+	children: ReactNode;
+	onClose: () => void;
+	title: string;
+}
+
+const Modal: React.FC<ModalProperties> = ({ children, onClose, title }) => {
+	const handleOverlayClick = useCallback(
+		(clickEvent: React.MouseEvent<HTMLDivElement>): void => {
+			if (clickEvent.target === clickEvent.currentTarget) {
+				onClose();
+			}
+		},
+		[onClose],
+	);
+
+	const handleOverlayKeyDown = useCallback(
+		(keyEvent: React.KeyboardEvent<HTMLDivElement>): void => {
+			if (
+				keyEvent.key === "Escape" ||
+				keyEvent.key === "Enter" ||
+				keyEvent.key === " "
```

Comment:
I don't think it's a good idea to add Enter and Space for closing. Modals will contain inputs and submit buttons and it can cause conflicts